### DISCUSSION TITLE
Using custom OSRM-Backend with maplibre-gl-directions

### DISCUSSION THREAD

**Discussion by deancochran** [maplibre/maplibre-gl-directions#199] [Q&A]
Hey there!

I've recently begun a side project in which I wanted to use Sveltekit, Maplibre, and OSRM to build a similar app like [BRouter](https://brouter.de/brouter-web/). I stumbled on this repo and was pleasantly surprised with how much has been implemented. I also attempted to join the slack community for Maplibre, but the documentation link to join is expired. So I have been toying with the capabilities of the package for a few weeks or so.

I'll leave my question here for now but will continue to work on my learning. 

Is there a way to adjust the routing engine to use a custom OSRM routing engine? I have managed to structure a docker container to expose the localhost on port 5000. I'm able to receive directions from the engine by [calling the backend](http://project-osrm.org/docs/v5.5.1/api/#general-options) (GET http:localhost:5000/route/v1), and I want to use maplibre to display the routes generated from my osrm backend.

**Comment by wipfli** [comment]
https://slack.openstreetmap.us should work. Can you update the readme in a pull request?

**Comment by smellyshovel** [comment]
Hello, @deancochran.

If I understood your question correctly, you just need to set the `api` and `profile` [options](https://maplibre.org/maplibre-gl-directions/api/interfaces/MapLibreGlDirectionsConfiguration.html) in accordance with your needs.

**Reply by deancochran** [reply]
Hey @smellyshovel, so cool! Thanks for the fast reply, it works like a charm. I will be following along, and reporting anything I find along the way. Thanks again.
